engine/src/graphics/vkHelpers.cpp

```diff
@@ -1,234 +1,128 @@
-#ifdef GEN_NDEBUG
-	vk::StructureChain<vk::InstanceCreateInfo>
-#else
-	vk::StructureChain<vk::InstanceCreateInfo, vk::DebugUtilsMessengerCreateInfoEXT>
-#endif
-	makeInstanceCreateInfoChain( vk::ApplicationInfo const & appInfo,
-								std::vector< const char * > const & layers,
-								std::vector< const char * > const & extensions )
+	vk::StructureChain<vk::InstanceCreateInfo> makeInstanceCreateInfoChain(vk::ApplicationInfo const & appInfo, std::vector<const char *> const & layers, std::vector<const char *> const & extensions)
 	{
 
-#ifdef GEN_NDEBUG
 		// When in non-debug mode, use the InstanceCreateInfo for instance creation.
 		vk::StructureChain<vk::InstanceCreateInfo> const instanceCreateInfo({{}, &appInfo, layers, extensions});
-#else
-		// In debug mode, also use the debugUtilsMessengerCallback in instance creation
-		vk::DebugUtilsMessageSeverityFlagsEXT const severityFlags(vk::DebugUtilsMessageSeverityFlagBitsEXT::eWarning |
-																  vk::DebugUtilsMessageSeverityFlagBitsEXT::eError);
-
-		vk::DebugUtilsMessageTypeFlagsEXT const messageTypeFlags(vk::DebugUtilsMessageTypeFlagBitsEXT::eGeneral |
-																 vk::DebugUtilsMessageTypeFlagBitsEXT::ePerformance |
-																 vk::DebugUtilsMessageTypeFlagBitsEXT::eValidation);
 
-		vk::StructureChain<vk::InstanceCreateInfo, vk::DebugUtilsMessengerCreateInfoEXT> instanceCreateInfo(
-			{{}, &appInfo, layers, extensions}, {{}, severityFlags, messageTypeFlags, &vk::util::debugUtilsMessengerCallback});
-#endif
 		return instanceCreateInfo;
 	}
 
-	// Create DebugUtilsMessengerCreateInfoEXT for Vulkan debug messaging.
-	vk::DebugUtilsMessengerCreateInfoEXT makeDebugUtilsMessengerCreateInfoEXT()
-	{
-		return { {},
-				vk::DebugUtilsMessageSeverityFlagBitsEXT::eWarning | vk::DebugUtilsMessageSeverityFlagBitsEXT::eError,
-				vk::DebugUtilsMessageTypeFlagBitsEXT::eGeneral | vk::DebugUtilsMessageTypeFlagBitsEXT::ePerformance |
-					vk::DebugUtilsMessageTypeFlagBitsEXT::eValidation,
-				&vk::util::debugUtilsMessengerCallback };
-	}
-
-	// Check if the requested validation layers are supported by the Vulkan instance.
-	bool checkValidationLayerSupport(const std::vector<std::string>& validationLayers)
-	{
-#ifndef GEN_NDEBUG
-		auto availableLayers = vk::enumerateInstanceLayerProperties();
-
-		for (auto const& layerName : validationLayers)
-		{
-			bool layerFound = false;
-
-			for (auto const& layerProperties : availableLayers)
-			{
-				if (layerName == layerProperties.layerName)
-				{
-					layerFound = true;
-					break;
-				}
-			}
-
-			if (!layerFound)
-			{
-				return false;
-			}
-		}
-
-		return true;
-#else
-		return true;
-#endif
-	}
 
 	// Check if the required device extensions are supported by the given physical device.
-	bool checkDeviceExtensionSupport(vk::PhysicalDevice device, const std::vector<const char*>& deviceExtensions)
+	bool checkDeviceExtensionSupport(const vk::PhysicalDevice device, const std::span<const char *> & deviceExtensions)
```

Comment:
`const std::span<const char*>` is incorrect, you want `std::span<const char* const>`.